apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts

```diff
@@ -201,10 +205,28 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
             }
         });
 
+        this.updatePlanSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.responseSubscriptionId = response.subscriptionId;
+                if (response.duration === "YEARLY") {
+                    this.isLoading = true;
+                    this.subscriptionResponse = response;
+                    this.initializePayment(response);
+                } else {
+                    this.openCashfreeDialog(response?.redirectLink);
+                }
+            }
+        });
+
+
         this.updateSubscriptionPaymentIsSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
             if (response) {
                 this.isLoading = false;
-                this.router.navigate(['/pages/new-company/' + this.subscriptionId]);
+                if (this.isChangePlan) {
+                    this.router.navigate(['/pages/subscription'])
+                } else {
+                    this.router.navigate(['/pages/new-company/' + this.subscriptionId])
```

Comment:
this.router.navigate(['/pages/new-company/' + this.subscriptionId]);

suggestion
this.router.navigate([`/pages/new-company/${this.subscriptionId}`]);